shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -233,6 +236,15 @@ - (void)resetChannels {
 // Channels get a reference to the engine, and therefore need manual
 // cleanup for proper collection.
 - (void)setupChannels {
+  __block FlutterEngine* engine = self;
```

Comment:
This is still strongly capturing `self` even though you set a block local. You can use self directly in the block, it's no different.
Are you trying to prevent a retain cycle? I'm not sure how these handlers are retained, it might not. I don't have the code in front of me at the moment.

Reply:
Yeah. The messenger is actually dependent on `self` anyway, so I think it's fine to just capture self.